app/models/lookup_value.rb

```diff
@@ -37,6 +38,11 @@ def value=(val)
     end
   end
 
+  def hidden_value=(val)
+    lookup_key.hidden_value = val
+    lookup_key.save unless lookup_key.new_record?
```

Comment:
This is causing updates for every (overridden?) lookup key it seems on a host#edit save.  I'm seeing lots of UPDATE SQL queries in the log when saving which are amounting to no-op changes:

<pre>
2015-12-01T11:34:21 [sql] [D]    (0.2ms)  UPDATE "lookup_keys" SET "updated_at" = '2015-12-01 11:34:21.532348', "default_value" = '--- my default!
 | ...
 | ' WHERE "lookup_keys"."type" IN ('PuppetclassLookupKey') AND "lookup_keys"."id" = 110
</pre>

Where it is actually changing the default value, it does seem to work but rewrites the default value each time.

A couple extra points about this:

a) Should this use an autosave association, perhaps from lookup value to lookup key?  I'm not sure if that might impact other parts of the app though, e.g. the Puppet class or lookup key edit pages.
b) This shouldn't save until the LookupValue itself is saved.
c) Will failures of lookup_key.save be detected?